src/bank02.asm

```diff
@@ -5813,42 +5813,40 @@ drawDefaultStatusBar:
     ret                                                ;; 02:6f28 $c9
 
 drawHPOnStatuBar:
-    ld   DE, $06                                       ;; 02:6f29 $11 $06 $00
-    push DE                                            ;; 02:6f2c $d5
-    ld   B, $04                                        ;; 02:6f2d $06 $04
-    call clearStatusBarSection                         ;; 02:6f2f $cd $6b $6f
-    pop  DE                                            ;; 02:6f32 $d1
-    ld   A, [wHPHigh]                                  ;; 02:6f33 $fa $b3 $d7
-    ld   H, A                                          ;; 02:6f36 $67
-    ld   A, [wHPLow]                                   ;; 02:6f37 $fa $b2 $d7
-    ld   L, A                                          ;; 02:6f3a $6f
-    call drawNumberOnStatusBar                         ;; 02:6f3b $cd $77 $6f
-    ret                                                ;; 02:6f3e $c9
+    ld   D, $13 ; Mode/Max-digits to write             ;; 02:6f29 $06 $40
+    ld   E, $00 ; WRAM Offset, 2-bytes for alignment   ;; 02:6f2b $0e $00
```

Comment:
What about passing a full 16 address of wHPMPTileBuffer in DE? The caller already needs to have knowledge of the memory layout of the tile cache in WRAM, and it would simplify `drawLeftAlignedNumberInWRAM`.

Reply:
I think I'm out of space in that function unless I combine the the two sub-functions. I can do that, but it makes the drawLeftAlignedNumberInWRAM function have even more scope.

Reply:
I just decided to change it... if it is a convention, it's kinda silly. All the registers get mucked with regularly...

Reply:
I just utilized the fact that I could return BC and DE out of drawLeftAlignedNumberInWRAM and into requestVRAMStatusBarTransfer.

Fixed now and looks much cleaner, thanks!